HelpSource/Classes/Document.schelp

```diff
@@ -239,6 +239,20 @@ Task({
 )
 ::
 
+method:: save
+Save this Document.
+argument:: path
+An optional instance of link::Classes/String:: indicating the path to save the Document. If a path is not provided, the current path (if it has been previously saved or read) is used. After a successful save, this Document's path will be updated if needed.
```

Comment:
> After a successful save, this Document's path will be updated if needed.

where does that happen?

Reply:
On the IDE side in ```DocumentManager::doSaveAs``` which updates the path and then calls ```syncLangDocument```. It would be possible to set the path immediately for synchronous access, but this would make it wrong if the save fails for any reason.